Sort by column keys

harimohanraj89 opened the pull request:
### Issue

This PR allows users to fix the issues they encountered in #48.

### Context

As noted in #48, tablesort currently assumes that the columns in table headers match the columns in the table body of every row. This leads to arguably erratic behavior when tablesort is applied to tables that utilize colspans.

I believe #130 attempts to rectify this problem by calculating the appropriate column to sort by, accounting for varying colspans in every row. However, it appears this approach wasn't seen to completion as @tristen noted [here](https://github.com/tristen/tablesort/pull/130#issuecomment-574724309). There were also some interesting edge cases to be considered with this mathematical approach. What happens when a row doesn't have any cells lining up with the clicked header because of, say, a full-width cell in that row? Do we treat the row as empty, or do we use the value of the full-width cell in sorting by any of the columns?

### This PR's Approach

This PR takes a more configuration-heavy approach, adding support for a `data-sort-column-key` data attribute to both headers and cells. Users can configure exactly which cells in each row they wish to be used for sorting when the corresponding header is clicked. This allows for users to "fix" the behavior of their tablesorts even when using colspans, as they are free to manually assign the appropriate column keys to table cells to sort the table as they see fit.

### Documentation

I've taken a stab at updating the documentation too, but I'm not 100% sure how to test that I've done that correctly. If you can provide guidance on how to test it out, I'd be happy to fix it up if it's broken.

tristen commented on demo/index.md:
```diff
@@ -205,6 +205,67 @@ It is possible to automatically sort the table once you create a Tablesort insta
+<thead>
+  <tr>
+    <th >Product</th>
+    <th colspan=2 data-sort-column-key="price">Price</th>
```
`colspan=2` value should be quoted! `colspan="2"`

harimohanraj89 replied:
Whoops! 🤦‍♂

tristen commented on demo/index.md:
```diff
@@ -205,6 +205,67 @@ It is possible to automatically sort the table once you create a Tablesort insta
+<thead>
+  <tr>
+    <th>Product</th>
+    <th colspan=2 data-sort-column-key="price">Price</th>
```
Same here on `colspan`

tristen commented on demo/index.md:
```diff
@@ -205,6 +205,67 @@ It is possible to automatically sort the table once you create a Tablesort insta
+
+This example sorts products by price, even though the prices are not in the same column as their header.
+
+{% highlight html %}<table id='column-keys' class='sort'>
```
Maybe leave out the `id` attribute here? duplicate ids can be finicky even though this is wrapped in highlight tags

tristen commented on demo/index.md:
```diff
@@ -205,6 +205,67 @@ It is possible to automatically sort the table once you create a Tablesort insta
+</table>
+{% endhighlight %}
+
+<table id='column-keys' class='sort'>
```
I know the documentation of this project is a little horrendous but for your code example to work, can you add 

```js
new Tablesort(document.getElementById('column-keys'));
``` 

Within the `<script>` tag [here](https://github.com/tristen/tablesort/pull/181/files#diff-46c1f60845b5267148ebc4d1a8f62654R282)?

harimohanraj89 replied:
Ah, I incorrectly assumed that the script at the bottom would initiate `Tablesort`s on all `table.sort` elements in here. My mistake!

I've given your suggestion a whirl!

harimohanraj89 commented:
Thanks for the review, @tristen! I figured the Travis failures were unrelated, good to know!

I've addressed your feedback, let me know if there's anything else, thanks!

tristen commented:
Published to npm as [`v5.2.0`](https://www.npmjs.com/package/tablesort)
